src/ice_ridge.F90

```diff
@@ -292,231 +322,248 @@ subroutine ice_ridging(IST, G, IG, mca_ice, mca_snow, mca_pond, TrReg, CS, US, d
 
   if (present(rdg_rate)) rdg_rate(:,:)=0.0
   do j=jsc,jec; do i=isc,iec
-  if ((G%mask2dT(i,j) .gt. 0.0) .and. (sum(IST%part_size(i,j,1:nCat)) .gt. 0.0)) then
-  ! feed locations to Icepack's ridge_ice
-
-    ! start like we're putting ALL the snow and pond in the ocean
-    IST%snow_to_ocn(i,j) = IST%snow_to_ocn(i,j) + sum(mca_snow(i,j,:))
-    IST%enth_snow_to_ocn(i,j) = IST%enth_snow_to_ocn(i,j) + sum(mca_snow(i,j,:)*TrReg%Tr_snow(1)%t(i,j,:,1))
-    IST%water_to_ocn(i,j) = IST%water_to_ocn(i,j) + sum(mca_pond(i,j,:))
-    aicen(1:nCat) = IST%part_size(i,j,1:nCat)
-
-
-    if (sum(aicen) .eq. 0.0) then ! no ice -> no ridging
-      IST%part_size(i,j,0) = 1.0;
-    else
-      ! set up ice and snow volumes
-      vicen(1:nCat) = mca_ice(i,j,1:nCat) /Rho_ice * US%Z_to_m  ! volume per unit area of ice (m)
-      vsnon(1:nCat) = mca_snow(i,j,1:nCat)/Rho_snow * US%Z_to_m ! volume per unit area of snow (m)
-
-      sh_Dt = (G%dyT(i,j)*G%IdxT(i,j)*(G%IdyCu(I,j) * IST%u_ice_C(I,j) - &
-                                       G%IdyCu(I-1,j)*IST%u_ice_C(I-1,j)) - &
-               G%dxT(i,j)*G%IdyT(i,j)*(G%IdxCv(i,J) * IST%v_ice_C(i,J) - &
-                                       G%IdxCv(i,J-1)*IST%v_ice_C(i,J-1)))
-      sh_Dd = (G%IareaT(i,j)*(G%dyCu(I,j) * IST%u_ice_C(I,j) - &
-                              G%dyCu(I-1,j)*IST%u_ice_C(I-1,j)) + &
-               G%IareaT(i,j)*(G%dxCv(i,J) * IST%v_ice_C(i,J) - &
-                              G%dxCv(i,J-1)*IST%v_ice_C(i,J-1)))
-
-      del_sh = sqrt(sh_Dd**2 + 0.25 * (sh_Dt**2 + &
-                   (0.25 * ((sh_Ds(I-1,J-1) + sh_Ds(I,J)) + &
-                            (sh_Ds(I-1,J) + sh_Ds(I,J-1))))**2 ) )*US%s_to_T ! H&D eqn 9
-      rdg_conv  = -min(sh_Dd,0.0)*US%s_to_T              ! energy dissipated by convergence ...
-      rdg_shear = 0.5*(del_sh-abs(sh_Dd))*US%s_to_T      ! ... and by shear
-
-      aice0 = IST%part_size(i,j,0)
-      if (aice0<0.) then
-         call SIS_error(WARNING,'aice0<0. before call to ridge ice.')
```

Comment:
Could we keep this warning? It would be good to know if we have negative ocean area.

Reply:
oh, so it is "okay" to have negative ocean area? 
Weirdly, I have a run I am debugging right now where the part_size(0)<0 in the checksum but this aice0<0 warning is not showing up. 
Should I switch to this branch and ridging scheme for the ARC12?